### ISSUE TITLE
Check max upload size and sever space availability before uploading a new file 

### ISSUE THREAD

**Issue by rperezb** [owncloud/ios#687]
AC:
- Before uploading a new file it has to be checked whether the file is not bigger than the max upload size or the available space on the server
- In case it will be bigger, the upload will be cancelled and the user will be warned with an error

**Comment by nasli** [comment]
The current error show "Waiting for server connection" (even if the server is connected). 
This will replace with an accurate error message.

**Comment by davivel** [comment]
@michaelstingl , is this still a green ticket?

**Comment by michaelstingl** [comment]
I don't think so. @butonic You raised priority. Please ping me to let me know the urgency.

**Comment by davivel** [comment]
Lack of quota in uploads was already handled in https://github.com/owncloud/ios/issues/822 , though **after** the upload instead of **before**. That should cover the 'urgency' of the topic, so this can go to the release after.
